src/libstd/fs.rs

```diff
@@ -2261,11 +2282,41 @@ mod tests {
     }
 
     #[test]
+    fn concurrent_recursive_mkdir() {
+        for _ in 0..50 {
+            let mut dir = tmpdir().join("a");
+            for _ in 0..100 {
```

Comment:
Looks like you reduced the number of times this test is run rather than the the length of the path 😉.